src/manager/msFocusManager.ts

```diff
@@ -74,19 +76,16 @@ export class MsFocusManager extends MsManager {
         }
 
         let actor = keyFocus;
-        let isChildrenOfMsWindow = false;
-        while (actor.get_parent() && !isChildrenOfMsWindow) {
+        while (actor.get_parent()) {
             actor = actor.get_parent();
             if (actor instanceof MsWindow) {
-                isChildrenOfMsWindow = true;
+                this.setFocusToMsWindow(actor);
+                return;
```

Comment:
Return is enough to break the while?

Reply:
Yes? It will return from the whole function.